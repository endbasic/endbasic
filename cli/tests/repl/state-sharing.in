' EndBASIC
' Copyright 2020 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Tests interactions between the stored program and the machine state.
'
' WARNING: This file contains literal control codes to interact with the
' full-screen editor.  See the README.md file for details on how to generate
' its corresponding .out file.

a = 3
PRINT a
CLEAR
PRINT a

EDIT ' Insert stored program until an escape character.
DIM a(1) AS INTEGER
a(0) = 123
PRINT "a(0) is"; a(0)


DIM a(1) AS STRING
a(0) = "Will not conflict"
RUN
RUN
RUN

PRINT "a(0) before CLEAR is"; a(0)
CLEAR ' Reset variables but not the stored program.
PRINT "a(0) after CLEAR is"; a(0)
RUN ' Still runs the program.

PRINT "a(0) before NEW is"; a(0)
NEW ' Reset everything.
PRINT "a(0) after NEW is"; a(0)
RUN ' Doesn't do anything.
