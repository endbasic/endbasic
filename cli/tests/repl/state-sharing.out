
    Welcome to EndBASIC X.Y.Z.

    Type HELP for interactive usage information.
    For a guided tour, type: LOAD "DEMOS:/TOUR.BAS": RUN

3
ERROR: Undefined variable a
[?1049h[?25l[38;5;15m[2J[1;1H[24;1H[48;5;4m ESC Exit                                             | <NO NAME> | Ln 1, Col 1 [49m[1;1H
[1;1H[?25hD[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 2 [49m[1;2H[?25hI[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 3 [49m[1;3H[?25hM[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 4 [49m[1;4H[?25h [?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 5 [49m[1;5H[?25ha[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 6 [49m[1;6H[?25h([?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 7 [49m[1;7H[?25h1[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 8 [49m[1;8H[?25h)[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 1, Col 9 [49m[1;9H[?25h [?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 10 [49m[1;10H[?25hA[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 11 [49m[1;11H[?25hS[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 12 [49m[1;12H[?25h [?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 13 [49m[1;13H[?25hI[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 14 [49m[1;14H[?25hN[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 15 [49m[1;15H[?25hT[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 16 [49m[1;16H[?25hE[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 17 [49m[1;17H[?25hG[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 18 [49m[1;18H[?25hE[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 19 [49m[1;19H[?25hR[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 1, Col 20 [49m[1;20H[?25h[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 1 [49m[2;1H[?25ha[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 2 [49m[2;2H[?25h([?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 3 [49m[2;3H[?25h0[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 4 [49m[2;4H[?25h)[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 5 [49m[2;5H[?25h [?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 6 [49m[2;6H[?25h=[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 7 [49m[2;7H[?25h [?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 8 [49m[2;8H[?25h1[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 2, Col 9 [49m[2;9H[?25h2[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 2, Col 10 [49m[2;10H[?25h3[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 2, Col 11 [49m[2;11H[?25h[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 1 [49m[3;1H[?25hP[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 2 [49m[3;2H[?25hR[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 3 [49m[3;3H[?25hI[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 4 [49m[3;4H[?25hN[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 5 [49m[3;5H[?25hT[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 6 [49m[3;6H[?25h [?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 7 [49m[3;7H[?25h"[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 8 [49m[3;8H[?25ha[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 3, Col 9 [49m[3;9H[?25h([?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 10 [49m[3;10H[?25h0[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 11 [49m[3;11H[?25h)[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 12 [49m[3;12H[?25h [?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 13 [49m[3;13H[?25hi[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 14 [49m[3;14H[?25hs[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 15 [49m[3;15H[?25h"[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 16 [49m[3;16H[?25h;[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 17 [49m[3;17H[?25h [?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 18 [49m[3;18H[?25ha[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 19 [49m[3;19H[?25h([?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 20 [49m[3;20H[?25h0[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 21 [49m[3;21H[?25h)[?25l[24;1H[48;5;4m ESC Exit                                           | <NO NAME>* | Ln 3, Col 22 [49m[3;22H[?25h[?25l[24;1H[48;5;4m ESC Exit                                            | <NO NAME>* | Ln 4, Col 1 [49m[4;1H[?25h[?1049l[39ma(0) is 123
a(0) is 123
a(0) is 123
a(0) before CLEAR is 123
ERROR: Unknown function or array a
a(0) is 123
a(0) before NEW is 123
ERROR: Unknown function or array a
End of input by CTRL-D
